karapace/karapacemetrics.py

```diff
@@ -0,0 +1,169 @@
+"""
+karapace - metrics
+Supports collection of system metrics
+list of supported metrics:
+connections-active - The number of active HTTP(S) connections to server.
+                     Data collected inside aiohttp request handler.
+
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from datetime import datetime
+from kafka.metrics import MetricName, Metrics
+from kafka.metrics.measurable_stat import AbstractMeasurableStat
+from kafka.metrics.stats import Avg, Max, Rate, Total
+from karapace.config import Config
+from karapace.statsd import StatsClient
+from typing import Dict, Optional
+
+import schedule
+import threading
+import time
+
+
+class Value(AbstractMeasurableStat):
+    """
+    An AbstractSampledStat that maintains a simple average over its samples.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.value = 0.0
+
+    # pylint: disable=unused-argument
+    def measure(self, config: object, now: int) -> float:
+        return self.value
+
+    def record(self, config: object, value: float, time_ms: int) -> None:
+        self.value = value
+
+
+class Singleton(type):
+    _instances: Dict["Singleton", "Singleton"] = {}
+
+    def __call__(cls, *args: str, **kwargs: int) -> "Singleton":
+        if cls not in cls._instances:
+            instance = super().__call__(*args, **kwargs)
+            cls._instances[cls] = instance
+        return cls._instances[cls]
+
+
+class KarapaceMetrics(metaclass=Singleton):
+    def __init__(self) -> None:
+        self.active: Optional[object] = None
+        self.stats_client: Optional[StatsClient] = None
+        self.is_ready = False
+        self.metrics = Metrics()
+        self.event = threading.Event()
```

Comment:
```suggestion
        self.stop_event = threading.Event()
```

suggestion: This can have more descriptive name.

Reply:
done